### PR TITLE
Rabbitmq additions

### PR DESCRIPTION
**Pull request by github-abcde**
### What does this PR do?
It adds states (in `states/rabbitmq_upstream.py` and functions (in `modules/rabbitmq.py`) to manipulate rabbitMQ upstream definitions.

### What issues does this PR fix or reference?
None that I'm aware of

### Previous Behavior
Unable to add/remove/modify rabbitMQ upstreams.

### New Behavior
RabbitMQ upstream definitions can be made `_present` or `_absent` or manipulated via execution functions.

### Tests written?
Yes

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/rabbitmq.py [hunk 1]
```diff
@@ -256,6 +257,33 @@ def list_vhosts(runas=None):
...
 
 
+def list_upstreams(runas=None):
+    '''
+    Returns a dict of upstreams based on rabbitmqctl list_parameters.
+
+    :param str runas: The name of the user to run this command as.
+
+    CLI Example:
+    .. code-block:: bash
```
Comment (mchugh19) [inline salt/modules/rabbitmq.py]: This needs an empty newline before and after

File: salt/modules/rabbitmq.py [hunk 2]
```diff
@@ -286,6 +314,24 @@ def vhost_exists(name, runas=None):
...
 
+def upstream_exists(name, runas=None):
+    '''
+    Return whether the upstreamexists based on rabbitmqctl list_parameters.
+
+    :param str name: The name of the upstream to check for.
+    :param str runas: The name of the user to run the command as.
+
+    CLI Example:
+    .. code-block:: bash
```
Comment (mchugh19) [inline salt/modules/rabbitmq.py]: Needs a newline before and after

File: salt/modules/rabbitmq.py [hunk 3]
```diff
@@ -1050,5 +1128,130 @@ def disable_plugin(name, runas=None):
...
+        when connecting to old brokers which determine queue HA mode using this
+        argument. Default is ``None``, meaning the queue is not HA.
+    The following parameter applies to federated queues only:
+    :param str queue: The name of the upstream queue. Default is to use the same
+        name as the federated queue.
+
+    :param str runas: The name of the user to run the command as.
+
+    CLI Example:
+    .. code-block:: bash
```
Comment (mchugh19) [inline salt/modules/rabbitmq.py]: newline before and after

**Comment by github-abcde** [comment]
@mchugh19 Could you review this again? It feels like this MR is in limbo somewhat, waiting for review but on nobody's radar (and I can't seem to find a way to find the relevant people to request for review otherwise).

**Comment by mchugh19** [comment]
I know the feeling. Unfortunately I'm just a contributor, and don't have any merge powers. I too have a few PRs outstanding with no activity right now. I think folks are currently busy on a test improvement initiative for the 2019 point release and upcoming neon.

@KChandrashekhar any thoughts?

**Comment by Ch3LL** [comment]
im going to be creating the neon branch today, i've tagged for this one to get migrated to that branch after the branch creation to ensure its included in the upcoming neon release. i'll try to make some time tomorrow or monday to review this one so we can make some progress on this one for ya @github-abcde

**Comment by Ch3LL** [comment]
FYI I migrated this PR from develop to neon to ensure it is included in the upcoming neon release. Let me know if this caused any issues. Thanks

**Comment by github-abcde** [comment]
@Ch3LL That's ok, I have local patches for the version of salt I'm running :)
Thanks for picking this up.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this looks great. Just have one small request. Do you think you could add a new note about this new feature to ./doc/topics/releases/neon.rst

Would love to have this highlighted in the release notes for neon

**Comment by Ch3LL** [comment]
also it looks like there is a doc test failure. pretty much you just need to add a rst file for you new state module. you can look at `doc/ref/states/all/salt.states.rabbitmq_user.rst` for an example. And if you could add it to the list here: doc/ref/states/all/index.rst

**Comment by github-abcde** [comment]
@Ch3LL Added the requested docs, and fixed up the docstring in `states/rabbitmq_upstream:present`.

**Comment by Ch3LL** [comment]
Updating the branch . The windows tests are not running, I believe that issues is fixed.

**Review by Ch3LL — APPROVED** [review]

**Comment by github-abcde** [comment]
Closing because https://github.com/saltstack/salt/pull/55767 is already merged.

**Pull request closed without merging**